internal/discovery/secret_lister.go

```diff
@@ -6,27 +6,47 @@ import (
 	listersv1 "k8s.io/client-go/listers/core/v1"
 )
 
+type SecretListerer interface {
+	Lister(namespace string) (listersv1.SecretNamespaceLister, bool)
+}
+
 type SecretListerConfig struct {
 	// Namespace can be used to restric the search to a particular namespace.
-	Namespace string
+	Namespaces []string
 	// Client is the Kubernetes client.Interface used to build informers.
 	Client kubernetes.Interface
 }
 
+type MultiNamespaceSecretListerer struct {
+	listers map[string]listersv1.SecretNamespaceLister
+}
+
+func (l MultiNamespaceSecretListerer) Lister(namespace string) (listersv1.SecretNamespaceLister, bool) {
```

Comment:
Right now we are creating a collection of listers, indexed by namespace, and returning it. That doesn't sound bad, but I wonder if returning a listerer capable of creating listers lazily would have any advantages.

This is, if we make this method to first check if there is a lister already for a naespace, and create it if there isn't.

WDYT about this approach? I'm not positive it's a better option, just wondering about it.

Reply:
this was the most simple solution i imaged ( not even mutex for the map :) ) since we have all namespaces in the config ( no need to dynamically build listers ), and unless there is use case for that I would keep it like this.